service/dap/server.go

```diff
@@ -561,6 +561,8 @@ func (s *Server) handleRequest(request dap.Message) {
 		}()
 		<-resumeRequestLoop
 	//--- Synchronous requests ---
+	// TODO(polina): target might be running when remote attach debug session
+	// is started. Support handling initialize and attach requests while running.
```

Comment:
There are more moving parts to supporting this feature than just this handler plus tests (see TODOs). Besides being able to enter initialize and attach handlers, we need to halt the process, which needs to work with new auto-resume features that Suzy is about to commit. That's why  prefer to send this out separately. 

Adding error checking for clients who send unexpected request sequences is out of scope of this PR. I have been going back and force on this one in general. I guess I can keep the TODO in onInitializeRequest that I removed with this change to reconsider this in case of duplicate initialize requests, but then we should protect other handlers from being called out of turn as well.

Reply:
Leaving this as a TODO seems like a good choice to me. This will limit the merging difficulty for whichever PR gets merged second, and we can add this later once we can see both working together.

Reply:
TODO is fine. 

For initialize request serving - that's purely for capability negotiation if I understand DAP correctly. I don't think we need to halt the process yet though and probably can be done as soon as a connection opens but before switching to general handleRequest (which then treat initialize request as an invalid request). But I guess that's a bigger change.